jsonschema-maven-plugin/README.md

```diff
@@ -0,0 +1,133 @@
+# Java JSON Schema Generator – Maven Plugin
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.github.victools/jsonschema-module-javax-validation/badge.svg)](https://maven-badges.herokuapp.com/maven-central/com.github.victools/jsonschema-maven-plugin)
+
+Maven plugin for the [jsonschema-generator](../jsonschema-generator) – Integrating JSON Schema generation in your builds
+
+## Features
+1. Generate the JSON schema for a Java class.
+2. Configure the generator by selecting the modules to be used.
+3. Configure the options of the modules used in the generation.
+4. Configure the use of custom modules.
+
+## Usage
+### Plugin definition
+```xml
+<plugin>
+    <groupId>com.github.victools</groupId>
+    <artifactId>jsonschema-maven-plugin</artifactId>
+    <version>4.8.0</version>
+    <executions>
+        <execution>
+            <goals>
+                <goal>schema-generator</goal>
```

Comment:
Maybe `generate` or `generate-schema` are nicer names for a `goal`?

How does this work exactly? When is the schema being generated? During each build (since some contained class might have been changed)? Or during what phase is this triggered by default?
I assume the phase can be configured, but can this task be skipped explicitly via some argument, e.g. `schema.skip` or something like that?

Reply:
- True. Changed goal to `generate-schema`
- Currently the generation is done in the `PROCESS_TEST_SOURCES` phase. I am not very sure about that. I selected it because is the first phase after the `COMPILE` phase in the phase list for a `jar`.
In case you want to skip the generation of the schema I propose to use the Maven profiling mechanism. In this setup you put the plugin definition is a profiled section.
This is standard Maven technology. I could spend some words on it in the Readme. Let's think about this.

Reply:
I have moved it to the Compile phase now. Let's assume that the java compilation will always be the first step in that phase.